Modules/Loadable/Markups/MRML/vtkMRMLMarkupsPlaneNode.cxx

```diff
@@ -0,0 +1,659 @@
+{
+  if (!normal)
+    {
+    return;
+    }
+
+  if (this->GetNumberOfControlPoints() < 3)
+    {
+    return;
+    }
+
+  this->GetNormal(normal);
+
+  vtkMRMLTransformNode* transformNode = this->GetParentTransformNode();
+  if (!transformNode)
+    {
+    return;
+    }
+
+  // Get transform
+  vtkNew<vtkGeneralTransform> transformToWorld;
+  transformNode->GetTransformToWorld(transformToWorld.GetPointer());
+
+  // Convert coordinates
+  double origin[3] = { 0 };
+  this->GetOrigin(origin);
+  transformToWorld->TransformVectorAtPoint(origin, normal, normal);
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLMarkupsPlaneNode::SetNormal(const double normal[3])
+{
+  if (!normal)
+    {
+    return;
+    }
+
+  MRMLNodeModifyBlocker blocker(this);
+  this->CreateValidPlane();
+
+  double newNormal[3] = { normal[0], normal[1], normal[2] };
+  vtkMath::Normalize(newNormal);
+
+  double currentNormal[3] = { 0 };
+  this->GetNormal(currentNormal);
+
+  double epsilon = 0.0001;
+  if (vtkMath::Dot(newNormal, currentNormal) >= 1.0 - epsilon)
+    {
+    // Normal vectors are equivalent, no change required.
+    return;
+    }
+
+  double angleRadians = vtkMath::AngleBetweenVectors(newNormal, currentNormal);
+  double rotationAxis[3] = { 0 };
+  vtkMath::Cross(currentNormal, newNormal, rotationAxis);
+  if (vtkMath::Norm(rotationAxis) < epsilon)
+    {
+    // New + old normals are facing opposite directions.
+    // Find a perpendicular axis to flip around.
+    vtkMath::Perpendiculars(currentNormal, rotationAxis, nullptr, 0);
+    }
+
+  vtkNew<vtkTransform> transform;
+  double point0[3] = { 0 };
+  this->GetNthControlPointPosition(0, point0);
+  transform->Translate(point0);
+  transform->RotateWXYZ(vtkMath::DegreesFromRadians(angleRadians), rotationAxis);
+  for (int i = 0; i < 3; ++i)
+    {
+    point0[i] = -1 * point0[i];
+    }
+  transform->Translate(point0);
+
+  double point1[3] = { 0 };
+  this->GetNthControlPointPosition(1, point1);
+  transform->TransformPoint(point1, point1);
+  this->SetNthControlPointPosition(1, point1[0], point1[1], point1[2]);
+
+  double point2[3] = { 0 };
+  this->GetNthControlPointPosition(2, point2);
+  transform->TransformPoint(point2, point2);
+  this->SetNthControlPointPosition(2, point2[0], point2[1], point2[2]);
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLMarkupsPlaneNode::SetNormalWorld(const double inNormal[3])
+{
+  double normal[3] = { inNormal[0], inNormal[1], inNormal[2] };
+
+  vtkMRMLTransformNode* transformNode = this->GetParentTransformNode();
+  if (transformNode)
+    {
+    // Get transform
+    vtkNew<vtkGeneralTransform> transformToWorld;
+    transformNode->GetTransformFromWorld(transformToWorld.GetPointer());
+
+    // Convert coordinates
+    double origin[3] = { 0 };
+    this->GetOriginWorld(origin);
+    transformToWorld->TransformVectorAtPoint(origin, normal, normal);
```

Comment:
I've tried to apply a non-linear transform to a plane and found it confusing that the 3 control points are no longer on the plane. I think a more intuitive way to apply a non-linear transform would be to apply the transform to the 3 control points and then fit a plane on those.